js/models/conversations.js

```diff
@@ -656,10 +657,13 @@
           profileKey = storage.get('profileKey');
         }
 
+        const loadData = Attachment.loadData(migrationContext.readAttachmentData);
+        const attachmentsWithData =
+            await Promise.all(messageWithSchema.attachments.map(loadData));
```

Comment:
I tried to surgically add places where we have to upgrade messages or load attachment metadata. I’d appreciate you thinking about places that I might have missed (and are therefore _not_ in this PR) based on your extensive experience with the codebase — thanks  👀🙇